Fix #16585 - BLOB JPG preview thumbnail

joltcoke opened the pull request:
Thumbnails are broken because imagecreatetruecolor() is expecting int but the params are either strings (from query string) or float. This PR fixes both issues.

Addresses the broken thumbnail mentioned in #16585 

Fixes: #16585

joltcoke committed:
Fix BLOB JPG preview thumbnail

Signed-off-by: Florian Schirmer <jolt@tuxbox.org>

williamdes commented:
Hi @joltcoke 
Could you rebase onto QA_5_1 ?
And change the PR target using the edit button ?

joltcoke commented:
Hi @williamdes 

i've rebased to QA_5_1 and converted the casts into intval() conversions to be consistent with the other code in the same file. This also fixed the linting issues.

williamdes committed:
Fix "Uncaught TypeError: Argument 1 passed to PhpMyAdmin\Core::downloadHeader()"

Fatal error: Uncaught TypeError: Argument 1 passed to PhpMyAdmin\Core::downloadHeader() must be of the type string, null given,
called in libraries/classes/Controllers/TransformationWrapperController.php on line 171 and defined in 
libraries/classes/Core.php:617 Stack trace: #0 
libraries/classes/Controllers/TransformationWrapperController.php(171): PhpMyAdmin\Core::downloadHeader(NULL, 'application/oct...') #1 
libraries/classes/Routing.php(92): PhpMyAdmin\Controllers\TransformationWrapperController->index(Array) #2 
index.php(18): PhpMyAdmin\Routing::callControllerForRoute('/transformation...', Object(FastRoute\Dispatcher\GroupCountBased), Object(Symfony\Component\DependencyInjection\ContainerBuilder)) #3 {main} thrown in 
libraries/classes/Core.php on line 617

Signed-off-by: William Desportes <williamdes@wdes.fr>

williamdes commented:
I tested your fix and it works great !
But I added one more commit because the `cn` parameter seems not to be sent and it did throw and error.

williamdes commented on libraries/classes/Controllers/TransformationWrapperController.php:
```diff
@@ -194,11 +195,11 @@ public function index(): void
             if ($ratioWidth < $ratioHeight) {
-                $destWidth = $srcWidth / $ratioHeight;
-                $destHeight = $_REQUEST['newHeight'];
+                $destWidth = intval(round($srcWidth / $ratioHeight));
```
Let me know if https://stackoverflow.com/a/58712745/5155484 could be an issue here also
